DataManagementSystem/Agent/RequestOperations/ReTransfer.py

```diff
@@ -47,67 +52,144 @@ class ReTransfer( DMSRequestOperationsBase ):
 
   """
 
-  def __init__( self, operation = None, csPath = None ):
+  def __init__(self, operation=None, csPath=None):
     """c'tor
 
     """
     # # base class ctor
-    DMSRequestOperationsBase.__init__( self, operation, csPath )
-    # # gMonitor stuff
-    gMonitor.registerActivity( "FileReTransferAtt", "File retransfers attempted",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "FileReTransferOK", "File retransfers successful",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "FileReTransferFail", "File retransfers failed",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-
-  def __call__( self ):
+    DMSRequestOperationsBase.__init__(self, operation, csPath)
+
+  def __call__(self):
     """ reTransfer operation execution """
+
+    # Check whether the ES flag is enabled so we can send the data accordingly.
+    if self.rmsMonitoring:
+      self.rmsMonitoringReporter = MonitoringReporter(monitoringType="RMSMonitoring")
+    else:
+      # # gMonitor stuff
+      gMonitor.registerActivity("FileReTransferAtt", "File retransfers attempted",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("FileReTransferOK", "File retransfers successful",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("FileReTransferFail", "File retransfers failed",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+
     # # list of targetSEs
     targetSEs = self.operation.targetSEList
     # # check targetSEs for removal
     targetSE = targetSEs[0]
-    bannedTargets = self.checkSEsRSS( targetSE )
+    bannedTargets = self.checkSEsRSS(targetSE)
     if not bannedTargets['OK']:
-      gMonitor.addMark( "FileReTransferAtt" )
-      gMonitor.addMark( "FileReTransferFail" )
+      if self.rmsMonitoring:
+        for status in ["Attempted", "Failed"]:
+          self.rmsMonitoringReporter.addRecord({
+              "timestamp": int(Time.toEpoch()),
+              "host": Network.getFQDN(),
+              "objectType": "File",
+              "operationType": self.operation.Type,
+              "parentID": self.operation.OperationID,
+              "status": status,
+              "nbObject": 1
```

Comment:
that's wrong (it was wrong). Either you consider the operation to be attempted (and possibly failed) and then nbObject is 1. Or/and you count each files